packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx

```diff
@@ -194,7 +194,14 @@ export const LivenessCameraModule = (
       darkModePreference.removeEventListener('change', updateColorModeHandler);
       lightModePreference.addEventListener('change', updateColorModeHandler);
     };
-  }, [canvasRef, videoRef, videoStream, isStartView]);
+  }, [
+    canvasRef,
+    videoRef,
```

Comment:
We don't need to track the refs in the dependency array, their references maintain stability since we access `.current` inside the body of the hook
```suggestion
```

Reply:
I'll remove `canvasRef` but react gets mad if we take out videoRef, I think because the ref itself might change since it comes from the `useMediaStreamInVideo` hook 
https://github.com/aws-amplify/amplify-ui/blob/4393c78a590499391090abe925acc834c792b815/packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx#L131

Reply:
Ahh got it. For sure out of scope for this PR (and tbh think it will be a non-trivial change) but we should refactor to keep `videoRef` scoped to wherever it's used directly. Passing refs around leads to unexpected behaviors

Reply:
Yeah makes sense I'll add that to our backlog. It definitely made this like 10x harder to fix